Assets/PresetsPerFolder/PresetPostProcessor.cs

```diff
@@ -0,0 +1,219 @@
+        {
+            HashSet<string> folders = new HashSet<string>();
+            foreach (var asset in addedAssets)
+            {
+                // a new Preset has been added, we need to make sure to update the dependency for this folder
+                if (asset.EndsWith(".preset"))
+                {
+                    folders.Add(Path.GetDirectoryName(asset));
+                }
+            }
+
+            foreach (var asset in deletedAssets)
+            {
+                // a Preset has been removed, we need to make sure to update the dependency for this folder
+                if (asset.EndsWith(".preset"))
+                {
+                    folders.Add(Path.GetDirectoryName(asset));
+                }
+            }
+
+            foreach (var movedAsset in movedAssets)
+            {
+                // a Preset has been moved, we need to make sure to update the dependency for the previous and new folder
+                if (movedAsset.destinationAssetPath.EndsWith(".preset"))
+                {
+                    folders.Add(Path.GetDirectoryName(movedAsset.destinationAssetPath));
+                }
+
+                if (movedAsset.sourceAssetPath.EndsWith(".preset"))
+                {
+                    folders.Add(Path.GetDirectoryName(movedAsset.sourceAssetPath));
+                }
+            }
+
+            // Make sure we don't add the dependency update for no reason.
+            if (folders.Count != 0)
+            {
+                // The dependencies need to be updated outside of the AssetPostprocessor calls
+                // so we register the method to the next Editor update
+                EditorApplication.delayCall += () =>
+                {
+                    DelayedDependencyRegistration(folders);
+                };
+            }
+        }
+
+        /// <summary>
+        /// This method loads all Presets in each of the given folder path
+        /// and updates the CustomDependency hash based on the presets currently in that folder.
+        /// </summary>
+        static void DelayedDependencyRegistration(HashSet<string> folders)
+        {
+            foreach (var folder in folders)
+            {
+                var presetPaths =
+                    AssetDatabase.FindAssets("glob:\"**.preset\"", new[] { folder })
+                        .Select(AssetDatabase.GUIDToAssetPath)
+                        .Where(presetPath => Path.GetDirectoryName(presetPath) == folder)
+                        .OrderBy(a => a);
+
+                Hash128 hash = new Hash128();
+                foreach (var presetPath in presetPaths)
+                {
+                    hash.Append(presetPath);
+                }
+
+                AssetDatabase.RegisterCustomDependency($"PresetPostProcessor_{folder}", hash);
+            }
+
+            // We have to manually trigger a Refresh at that point
```

Comment:
Suggestion: "We have to manually trigger a Refresh so that the AssetDatabase triggers a dependency check on the folder hash we updated."